qe/drivers.py

```diff
@@ -79,6 +79,34 @@ def integral_category(i, j, k, l):
         return "G"
 
 
+#  ___  _________ _____
+#  |  \/  || ___ \_   _|
+#  | .  . || |_/ / | |
+#  | |\/| ||  __/  | |
+#  | |  | || |    _| |_
+#  \_|  |_/\_|    \___/
+#
+
+
+def get_chunk(comm: MPI.COMM_WORLD, psi_global: Psi_det):
+    # MPI function, compute work distribution and get chunk of determinants
+    # Used to distribute internal (diagonalization) or external space (PT2), depending on application
+    # TODO: Doc tests
+    global_size = len(psi_global)
+    rank = comm.Get_rank()
+    world_size = comm.Get_size()
+    # At job initialization, each rank computes distribution of determinants
+    floor, remainder = divmod(global_size, world_size)
+    ceiling = floor + 1
+    # Compute work distribution (size of chunk given to each node)
+    distribution = np.array([ceiling] * remainder + [floor] * (world_size - remainder), dtype="i")
+    # Compute offests (start of the local chunk)
+    offsets = np.zeros(world_size, dtype="i")
+    np.add.accumulate(distribution[:-1], out=offsets[1:])
+
```

Comment:
I'm a little lost of what this function is doing.
If it's just splitting the external space, we don't need any MPI call expect (`rank = comm.Get_rank()`)

Everybody computes psi_external (yes this is stupid for now but whatever) and then they just take the chunk they need.

I guess I don't understand the function :)